dhall/src/Dhall/Import.hs

```diff
@@ -1119,3 +1121,52 @@ instance Show ImportResolutionDisabled where
 assertNoImports :: MonadIO io => Expr Src Import -> io (Expr Src Void)
 assertNoImports expression =
     Dhall.Core.throws (traverse (\_ -> Left ImportResolutionDisabled) expression)
+
+{-| This function is used by the @--transitive@ flag of the
+    @dhall {freeze,format,lint}@ subcommands to determine which dependencies
+    to descend into
+-}
+dependencyToFile :: Status -> Import -> IO (Maybe FilePath)
+dependencyToFile status import_ = flip State.evalStateT status $ do
+    parent :| _ <- zoom stack State.get
+
+    child <- fmap chainedImport (hoist liftIO (chainImport parent import_))
+
+    let Import{ importHashed, importMode } = child
+
+    let ImportHashed{ importType } = importHashed
+
+    let ignore = return Nothing
+
+    -- Only descend into "normal" imports
+    case importMode of
+        RawText ->
+            ignore
+
+        Location ->
+            ignore
+
+        Code ->
+            case importType of
+                Local filePrefix file -> do
+                    let descend = liftIO $ do
+                            path <- localToPath filePrefix file
+
+                            return (Just path)
+
+                    -- Only follow relative imports when modifying dependencies
+                    case filePrefix of
+                        Here     -> ignore
+                        Parent   -> ignore
+                        Absolute -> descend
+                        Home     -> descend
```

Comment:
Actually, the intention was for it to be the other way around so this is a mitsake.  However, now that I'm thinking about this more closely I realized that this needs to be redone a bit because it is checking the file prefix of the fully chained path, rather than the relative path.

In other words, the reason it accidentally worked in my local testing was because the initial path I specified was home-anchored, so all subsequent chained paths were home-anchored and triggered the descent anyway because the check for descent was backwards.